src/images.rs

```diff
@@ -0,0 +1,254 @@
+use chrono::{DateTime, Utc};
+use crypto::digest::Digest;
+use crypto::sha1::Sha1;
+use std::collections::{HashMap, HashSet};
+use std::fs::{self, File};
+use std::io::{Read, Seek, SeekFrom, Write};
+use std::str::FromStr;
+use std::time::Duration;
+
+use datatype::{Error, Util};
+
+
+/// Metadata regarding an image to be transferred between ECUs in chunks.
+#[derive(Serialize, Deserialize)]
+pub struct ImageMeta {
+    pub image_name: String,
+    pub sha1sum: String,
+    pub num_chunks: u64,
+}
+
+impl ImageMeta {
+    pub fn new(image_name: String, sha1sum: String, num_chunks: u64) -> Self {
+        ImageMeta { image_name: image_name, sha1sum: sha1sum, num_chunks: num_chunks }
+    }
+}
+
+
+/// Read a local image in chunks for sending to a `Secondary` ECU.
+#[derive(Serialize, Deserialize)]
+pub struct ImageReader {
+    pub image_name: String,
+    pub image_dir: String,
+    pub image_size: u64,
+    pub chunk_size: u64,
```

Comment:
Actually, this should be fixed at something small like `u16` (maybe `u32` if assume all secondaries at an address space that big) because then you don't have the edge case where a reader and writer and have different `usize`s.